chat_exporter/construct/message.py

```diff
@@ -271,14 +304,31 @@ async def build_assets(self):
         for e in self.message.embeds:
             self.embeds += await Embed(e, self.guild).flow()
 
+        for snapshot in self.message.message_snapshots:
+            for se in snapshot.embeds:
+                self.embeds += await Embed(se, self.guild).flow()
+                self.message.reference = message_reference_forwarded
+
         for a in self.message.attachments:
             if self.attachment_handler and isinstance(self.attachment_handler, AttachmentHandler):
                 a = await self.attachment_handler.process_asset(a)
             self.attachments += await Attachment(a, self.guild).flow()
+        
+        for snapshot in self.message.message_snapshots:
```

Comment:
Can we not merge all of the `for snapshot in self.message.message_snapshots:` loops into one?